tests/Unit/Helpers/Evolution/Systems/GeneralizedHarmonic/GhDampingFunctions/TestHelpers.hpp

```diff
@@ -0,0 +1,104 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Framework/TestingFramework.hpp"
+
+#include <memory>
+#include <pup.h>
+#include <string>
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/GhDampingFunctions/GhDampingFunction.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/Overloader.hpp"
+
+namespace TestHelpers {
+namespace GhDampingFunctions {
+namespace detail {
+template <size_t VolumeDim, typename Fr, class... MemberArgs, class T>
+void check_impl(
+    const std::unique_ptr<GhDampingFunction<VolumeDim, Fr>>&
+        in_gh_damping_function,
+    const std::string& python_function_prefix, const T& used_for_size,
+    const std::array<std::pair<double, double>, 1> random_value_bounds,
+    const MemberArgs&... member_args) noexcept {
+  using GhDampingFunc = GhDampingFunction<VolumeDim, Fr>;
+  using CallOperatorFunction =
+      Scalar<T> (GhDampingFunc::*)(const tnsr::I<T, VolumeDim, Fr>&)
+          const noexcept;
+
+  const auto member_args_tuple = std::make_tuple(member_args...);
+  const auto helper =
+      [&](const std::unique_ptr<GhDampingFunc>& gh_damping_function) noexcept {
```

Comment:
can you capture things explicitly? I know it's a bit tedious, but there are edge cases where capturing everything can lead to really, really bad bugs